pkg/config/storage/lvmd.go

```diff
@@ -0,0 +1,39 @@
+package storage
+
+import (
+	"fmt"
+
+	"gopkg.in/yaml.v2"
+	"k8s.io/klog/v2"
+)
+
+const ForcedSocketDir = "/run/lvmd/lvmd.sock"
+
+// LvmdConfig holds the full structure of the lvmd configuration
+type LvmdConfig map[string]interface{}
+
+// UnmarshalYAML overrides the socket-dir when reading from yaml to LvmdConfig in order
+// to prevent users from breaking the CSIPlugin with missing or incorrect socket paths.
```

Comment:
Another safer way to do this would be to add a "device classes" field to the MicroShift configuration file. That avoids exposing the `socket-name` option, so the user can't break it, but gives them control over the values they need to be able to change. It also lets us validate those settings, so we could fail to start if there is a list of device classes but none has the name "default", for example.

Reply:
I'd done that initially in fact but there were a few things that bugged me with the approach:

- The lvmd process already validates the config, concisely logs validation errors, and crashes immediately.  So it's obvious that, when the lvmd pod is crashLooping, `oc logs` will produce less than 5 lines describing what's wrong.  If microshift were made to crash on lvmd validation errors, the cause is much less obvious with the amount of log spam the process generates.  

- copying the lvmd config API in microshift feels like bad practice, and it also implies reflecting how lvmd validates the config fields.  The lvmd version is pinned in microshift of course, but it would essentially create an invisible dependency on the lvmd code base.  Less horrible that making lvmd a direct dependency, admittedly.

- Users can view the lvmd configmap in the cluster, edit it, and test config changes live against the cluster.  It ends up being a much cleaner debugging approach.